apps/server/src/state.ts

```diff
@@ -132,49 +143,63 @@ export const stateMutations = {
         },
       );
     },
-    resume(timer: OntimeEvent, restorePoint: RestorePoint) {
+    resume(event: OntimeEvent, restorePoint: RestorePoint) {
       mutate((state) => {
-        state.timer.selectedEventId = timer.id;
-        state.timer.timeEnd = timer.timeEnd;
-        state.timer.pausedAt = restorePoint.pausedAt;
+        state.timer.clock = clock.timeNow();
 
-        state.timer.duration = calculateDuration(timer.timeStart, timer.timeEnd);
-        state.playback = restorePoint.playback;
-        state.timer.timerType = timer.timerType;
-        state.timer.endAction = timer.endAction;
+        // TODO: the duplication of timer data would not be necessary
+        // once event loader is merged here
+        state.timer.selectedEventId = event.id;
         state.timer.startedAt = restorePoint.startedAt;
-        state.timer.addedTime = restorePoint.addedTime;
+        state.timer.timeEnd = event.timeEnd;
+        state.timer.duration = calculateDuration(event.timeStart, event.timeEnd);
         state.timer.current = state.timer.duration;
 
+        state.timer.timerType = event.timerType;
+        state.timer.endAction = event.endAction;
+
+        state.playback = restorePoint.playback;
+        state.timer.pausedAt = restorePoint.pausedAt;
+        state.timer.addedTime = restorePoint.addedTime;
+
+        // check if event finished meanwhile
         if (state.timer.timerType === TimerType.TimeToEnd) {
-          const now = clock.timeNow();
-          state.timer.current = getCurrent(now, state.timer.duration, 0, 0, now, timer.timeEnd, state.timer.timerType);
+          state.timer.current = getCurrent(
+            state.timer.clock, // TODO: should this be startedAt?
```

Comment:
from what I've seen it seems like it

one thing we can think of is to refactor `getCurrent` to use named arguments since we have 7 arguments there, would be easier and less confusing when we need to call it